src/org/sosy_lab/common/configuration/ConfigurationBuilder.java

```diff
@@ -250,35 +253,72 @@ public ConfigurationBuilder loadFromFile(Path file)
    * @throws IllegalArgumentException If the resource cannot be found or read, or contains invalid
    *     syntax or #include directives.
    */
+  @SuppressWarnings("try")
   public ConfigurationBuilder loadFromResource(Class<?> contextClass, String resourceName) {
     URL url = Resources.getResource(contextClass, resourceName);
     CharSource source = Resources.asCharSource(url, StandardCharsets.UTF_8);
 
     setupProperties();
 
     // Get the path to the source, used for error messages and resolving relative path names.
-    @Var Path sourcePath;
-    @Var String sourceString;
     try {
-      sourcePath = Paths.get(url.toURI());
-      sourceString = sourcePath.toString();
-    } catch (URISyntaxException | FileSystemNotFoundException | IllegalArgumentException e) {
+      // try to access the requested file.
+      URI uri = url.toURI();
+      try (FileSystem fs = getFileSystemForUri(uri)) {
+        // Path uses FileSystemProvider internally to access the file, thus fs is unused.
+        Path sourcePath = Paths.get(uri);
+        parseSource(
+            contextClass, resourceName, source, Optional.of(sourcePath), sourcePath.toString());
+      }
+    } catch (URISyntaxException
+        | FileSystemNotFoundException
+        | IllegalArgumentException
+        | IOException e) {
       // If this fails, e.g., because url is a HTTP URL, we can also use the raw string.
       // This will not allow resolving relative path names, but everything else works.
-      sourcePath = null;
-      sourceString = url.toString();
+      parseSource(contextClass, resourceName, source, Optional.empty(), url.toString());
```

Comment:
Won't this still also the same kind of exceptions if we pass a string as source that was not able to be converted to a `Path` in this method?

Reply:
This was not addressed? I see no way how this could be expected to work when `Paths.get()` is called on the result of `url.toString()` later, given that the conversion via `URI` has already failed?

Reply:
From my point, all cases that resulted in an execution of that branch, are handled as before, except the case, where a JAR file is accessed.
Thus, if it failed before, it will also fail now.
Currently I have no test case where I run into this branch.
Could you give an example why this case is required?

Do we have any JUnit test for this code?

Reply:
Sure, the current code improves the situation, but now that we recognize that other cases will still fail for the same reason, we should fix these, too, right?

We cannot unit test this properly because it requires interfacing with Java APIs and use of URLs.

Reply:
Right, I thought the same: testing might not be possible.
Which further cases do you know that should be handled?
Can we write a documentation for manual testing at least?

Reply:
Actually, the following code might work for testing:
```
    URL url =
        new URL(
            "https://raw.githubusercontent.com/sosy-lab/cpachecker/trunk/config/includes/resource-limits.properties");
    CharSource source = Resources.asCharSource(url, StandardCharsets.UTF_8);
    Configuration.builder().loadFromSource(source, url.toString(), url.toString());
```
This uses `loadFromSource`, but this in turn calls the same `parse` method as `loadFromResource`, it is just easier to specify the parameters. Note that the last parameter is `toString()` of an URL, just like in the `catch` block of `loadFromResource`.

For me, this creates a `UnixPath` with the `https://...`, which happens to work. But I guess it would crash on Windows?

Reply:
Yes, `https://...` crashes on Windows.
And I could not find any fix, because there is no matching FileSystemProvider available.
By default, `FileSystemProvider.installedProviders()` only lists the providers for `[UnixFileSystemProvider, JrtFileSystemProvider]` or `[WindowsFileSystemProvider, JrtFileSystemProvider]`, depending on the operating system. Any further filesystem access, e.g., accessing zip-files or http-files on Linux, might not be specified as default by Java itself, but provided by the environment.

Thus, I would not expect further support from Java and stay with the minimal working solution, which is already implemented in this PR.

Reply:
Well, the fix could be to not rely on `Path` if it does not support what we need (as mentioned in https://github.com/sosy-lab/java-common-lib/issues/34#issuecomment-728745129).

This is somewhat tricky, because these `Path`s end up in public API, but at least we can avoid the crashes now and refactor some things to make all this more clear. I just pushed a commit that does this, please review (and ideally, test again on Windows).